frontend/services/cuda/CUDA_Service.cpp

```diff
@@ -0,0 +1,46 @@
+#include "CUDA_Service.hpp"
+
+#ifdef MM_CUDA_ENABLED
+
+#include <stdexcept>
+
+#include "cuda.h"
+
+#include "mmcore/utility/log/Log.h"
+
+
+static void log(std::string const& text) {
+    const std::string msg = "CUDA_Service: " + text;
+    megamol::core::utility::log::Log::DefaultLog.WriteInfo(msg.c_str());
+}
+
+
+bool megamol::frontend::CUDA_Service::init(void* configPtr) {
+    cuInit(0);
+    int deviceCount = 0;
+    cuDeviceGetCount(&deviceCount);
+    if (deviceCount == 0) {
+        throw std::runtime_error("[CUDA_Service]: No Cuda device available");
```

Comment:
Instead of exception: log error and return 0
This way the frontend can shut down other services gracefully before terminating MegaMol. 
There is nobody waiting to catch this exception, anyway.